src/types.jl

```diff
@@ -60,6 +51,22 @@ struct ManagedSCSMatrix{T<:SCSInt}
     end
 end
 
+function ManagedSCSMatrix{T}(m::Integer, n::Integer, A::SparseMatrixCSC) where T
+    # we convert everything to make sure that no conversion will
+    # accidentally happen when ccalling `new`, so that `scsmatref`
+    # holds pointers to actual data stored in the fields.
+
+    values = convert(Vector{Cdouble}, copy(A.nzval))
+    rowval = convert(Vector{T}, A.rowval .- 1)
+    colptr = convert(Vector{T}, A.colptr .- 1)
+
+    return ManagedSCSMatrix(m, n, values, rowval, colptr)
+end
+
+function ManagedSCSMatrix{T}(m::Integer, n::Integer, A::Matrix) where T
+    return ManagedSCSMatrix(m, n, sparse(A))
+end
```

Comment:
```suggestion
function ManagedSCSMatrix{T}(m::Integer, n::Integer, A::AbstractMatrix) where T
    return ManagedSCSMatrix{T}(m, n, sparse(A))
end
```
`Abstract` is not important, missing `T` is